pandas/_libs/hashtable.pxd

```diff
@@ -41,7 +41,7 @@ cdef class StringHashTable(HashTable):
 
 cdef struct Int64VectorData:
     int64_t *data
-    size_t n, m
+    ssize_t n, m
```

Comment:
Not sure about using ssize_t - does Py_size_t not work here instead?

Reply:
no, since one is signed and the other unsigned. look at
https://github.com/pandas-dev/pandas/blob/d054c037132b6cdb61cdf3ed90499287f163aeb4/pandas/_libs/hashtable_class_helper.pxi.in#L32-L34

The two are supposed to match, no ? ssize_t is good. it avoids undeflows in calculations, and the lost bit doesn't matter even on 32bit archs.

Reply:
Your comment about `ssize_t` would apply generally, but most of this code operates in the context of Python which made the design decision to use the signed `Py_size_t` for indexing instead. So if this touches anything for indexing (which I think it does) will want to keep that consistent

Reply:
`Py_size_t` is not even defined and  Python actually made the design choice to use Py_ssize_t everywhere.

If you mean use `Py_ssize_t` instead of `ssize_t `, ok, but that's a different issue.

Reply:
Yes

Reply:
done